tests/test_securitygroup.py

```diff
@@ -81,14 +81,68 @@ def test_misconfiguration(get_details, get_properties):
         security_group.create_security_group()
 
 
-@patch('foremast.securitygroup.create_securitygroup.boto3')
-@patch('foremast.securitygroup.create_securitygroup.get_properties')
-@patch('foremast.securitygroup.create_securitygroup.get_security_group_id')
-@patch("foremast.securitygroup.create_securitygroup.get_details")
+@mock.patch('foremast.securitygroup.create_securitygroup.boto3')
+@mock.patch('foremast.securitygroup.create_securitygroup.get_properties')
+@mock.patch('foremast.securitygroup.create_securitygroup.get_security_group_id')
+@mock.patch("foremast.securitygroup.create_securitygroup.get_details")
 def test_tags(get_details, get_security_group_id, get_properties, boto3):
     """Make bad Security Group definitions more apparent."""
     get_properties.return_value = {'security_group': {}}
     get_security_group_id.return_value = 'SGID'
 
     security_group = SpinnakerSecurityGroup()
     assert security_group.add_tags() is True
+
+
+@mock.patch('foremast.securitygroup.create_securitygroup.get_details')
+@mock.patch('foremast.securitygroup.create_securitygroup.get_properties')
+def test_default_security_groups(mock_properties, mock_details):
+    """Make sure default Security Groups are added to the ingress rules."""
+    ingress = {
+        'test_app': [
+            {
+                'start_port': 30,
+                'end_port': 30,
+            },
+        ],
+    }
+
+    mock_properties.return_value = {
+        'security_group': {
+            'ingress': ingress,
+            'description': '',
+        },
+    }
+
+    test_sg = {
+        'myapp': [
+            {
+                'start_port': '22',
+                'end_port': '22',
+                'protocol': 'tcp'
+            },
+        ]
+    }
+    with mock.patch.dict('foremast.securitygroup.create_securitygroup.DEFAULT_SECURITYGROUP_RULES', test_sg):
+        sg = SpinnakerSecurityGroup()
+        ingress = sg.update_default_rules()
+        assert 'myapp' in ingress
+
+
+@mock.patch('foremast.securitygroup.create_securitygroup.get_details')
+@mock.patch('foremast.securitygroup.create_securitygroup.get_properties')
+def test_securitygroup_references(mock_properties, mock_details):
+    """Make sure default Security Groups are added to the ingress rules."""
+    test_sg = {
+        '$self': [
+            {
+                'start_port': '22',
+                'end_port': '22',
+                'protocol': 'tcp'
+            },
+        ]
+    }
+
+    sg = SpinnakerSecurityGroup(app='myapp')
+    ingress = sg.resolve_self_references(test_sg)
+    assert 'myapp' in ingress
```

Comment:
Probably good to check the list of ports as well. Would hate to have that slip through by accident in the future.

Reply:
Updated test to check port.